Realm/Realm/DatabaseTypes/RealmCollectionBase.cs

```diff
@@ -311,28 +311,37 @@ private void OnChange(IRealmCollection<T> sender, ChangeSet? change)
 
                 var raiseAdded = TryGetConsecutive(change.InsertedIndices, i => this[i], out var addedItems, out var addedStartIndex);
 
-                if (raiseAdded || raiseRemoved)
-                {
-                    if ((raiseAdded && raiseRemoved) ||
-                        (raiseAdded && addedItems == null) ||
-                        (raiseRemoved && removedItems == null))
-                    {
-                        RaiseCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
-                        RaisePropertyChanged();
-                        return;
-                    }
+                var raiseReplaced = TryGetConsecutive(change.NewModifiedIndices, i => this[i], out var replacedItems, out var replacedStartIndex);
 
+                // Only raise specialized notifications if we have exactly one change type to report
```

Comment:
Could we compute the replaced notifications after added and removed so we don't need to raise Reset in case we have added + replaced or removed + replaced? 
With replaced the index should be stable, so I suppose it could work?